app/services/publish_reporting_service.rb

```diff
@@ -0,0 +1,93 @@
+class PublishReportingService
+  def initialize(recruitment_cycle_scope: RecruitmentCycle)
+    @courses = recruitment_cycle_scope.courses
+    @providers = recruitment_cycle_scope.providers
+  end
+
+  class << self
+    def call(recruitment_cycle_scope:)
+      new(recruitment_cycle_scope: recruitment_cycle_scope).call
+    end
+  end
+
+  def call
+    days_ago = 30.days.ago
```

Comment:
Code Climate is complaining that this method is over 25 lines long. I was on the fence about mentioning it until I saw the specs for the scopes below ... I'd break these values out into methods and separate the logic and tests out from the presentation here.